system/jlib/jlog.cpp

```diff
@@ -605,37 +605,108 @@ void LogMsg::fprintTable(FILE * handle, unsigned fields) const
         fprintf(handle, "%s%s%s%s\n", quote, prefix, text.str(), quote);
 }
 
+constexpr EnumMapping MsgFieldMap[] =
+{
+    { MSGFIELD_msgID,     "MsgID    " },
+    { MSGFIELD_audience,  "Audience " },
+    { MSGFIELD_class,     "Class    " },
+    { MSGFIELD_detail,    "Detail     " },
+    { MSGFIELD_date,      "Date       " },
+    { MSGFIELD_microTime, "Time(micro)     " },
+    { MSGFIELD_milliTime, "Time(milli)  " },
+    { MSGFIELD_time,      "Time        " },
+    { MSGFIELD_process,   "PID   " },
+    { MSGFIELD_thread,    "TID   " },
+    { MSGFIELD_session,   "SessionID           " },
+    { MSGFIELD_node,      "Node                " },
+    { MSGFIELD_job,       "JobID  " },
+    { MSGFIELD_user,      "UserID  " },
+    { MSGFIELD_component, "Compo " }
+};
+
+constexpr unsigned sizeMsgFieldMap = arraysize(MsgFieldMap);
+
 void LogMsg::fprintTableHead(FILE * handle, unsigned fields)
 {
-    if(fields & MSGFIELD_audience)
-        fprintf(handle, "Audience ");
-    if(fields & MSGFIELD_class)
-        fprintf(handle, "Class    ");
-    if(fields & MSGFIELD_detail)
-        fprintf(handle, "    Detail ");
-    if(fields & MSGFIELD_msgID)
-        fprintf(handle, "   MsgID ");
-    if(fields & MSGFIELD_date)
-        fprintf(handle, "      Date ");
-    if(fields & MSGFIELD_time)
-        fprintf(handle, "    Time ");
-    if(fields & MSGFIELD_process)
-        fprintf(handle, "  PID ");
-    if(fields & MSGFIELD_thread)
-        fprintf(handle, "  TID ");
-    if(fields & MSGFIELD_session)
-        fprintf(handle, "      SessionID      ");
-    if(fields & MSGFIELD_node)
-        fprintf(handle, "               Node ");
-    if(fields & MSGFIELD_job)
-        fprintf(handle, "  JobID ");
-    if(fields & MSGFIELD_user)
-        fprintf(handle, " UserID ");
-    if(fields & MSGFIELD_component)
-        fprintf(handle, " Compo ");
-    fprintf(handle, "\n\n");
+
+    fputs("#", handle);
+    for (unsigned i=0; i < sizeMsgFieldMap; ++i)
+        if (fields & MsgFieldMap[i].val)
+        {
+            if (MsgFieldMap[i].val == MSGFIELD_time && (fields & (MSGFIELD_microTime|MSGFIELD_milliTime)) )
+                continue;
+            fputs(MsgFieldMap[i].str, handle);
+        }
+    fputs("\n", handle);
+}
+
+#define MSG_HEADER_MAX_SIZE COUNT_MSGFIELD*10
```

Comment:
Not convinced by this...  Might be better to calculate it by adding up the sizes of the MsgFieldMap strs - since not time critical.  Would also mean COUNT_MSGFIELD isn't required.  (It also doesn't really matter if it is too large...)